spec/duration.html

```diff
@@ -2,44 +2,78 @@
 <meta charset="utf8" />
 
 <emu-clause id="sec-temporal-duration">
-    <h1>Duration Objects</h1>
-    <p>
-    Duration objects describe the difference between two like Date, Time, or DateTime objects.
-    They are only created via the <i>.minus()</i> methods of these objects.
-    </p>
+  <h1>Temporal.Duration Objects</h1>
+  <p>
+    A Temporal.Duration object describes the difference between two like Date,
+    Time, or DateTime objects. Objects of this type are only created via the
+    <i>.minus()</i> methods of these objects.
+  </p>
+  <ul>
+    <li>
+      The <i>.minus()</i> operation on Temporal.Date objects creates a
+      Temporal.Duration that only has years, months, and days properties.
+    </li>
+    <li>
+      The <i>.minus()</i> operation on Temporal.Time objects creates a
+      Temporal.Duration that only has hours, minutes, seconds, milliseconds,
+      microseconds, and nanoseconds properties.
+    </li>
+    <li>
+      The <i>.minus()</i> operation on Temporal.Instant objects creates a
+      Temporal.Duration like that of Time objects.
+    </li>
+    <li>
+      The <i>.minus()</i> operation on Temporal.DateTime objects creates a
+      Temporal.Duration that has all member properties, except if passed an
+      option indicating it should be created as a Time object or Date object.
+    </li>
+  </ul>
+  <p>
+    When created, the smaller units are combined into full larger units (i.e.
+    1000 milliseconds are recorded as 1 seconds) up to the largest available
+    unit. The result of this is that the largest available unit may have a
+    larger value in some circumstances. Durations created from Time objects may
+    have an hours property larger than 24, which would be invalid for a Duration
+    created from a DateTime object.
```

Comment:
Wouldn't it be easier and more consistent for all Duration objects to have the same fields? What's the rationale for it behaving differently?

Reply:
Also, specifically on "Durations created from Time objects may have an hours property larger than 24", how? The difference between the earliest valid time and the latest valid time would be 24 hours, so this isn't even an issue.

Reply:
Also also, how does this work with days and months? How can you combine a number of days together to get a month?

Reply:
If you take

duration = datetime.difference(datetime)

and then
time.plus(duration)

it should throw if the duration has more days,months,years

and for

date.plus(duration)

it should throw if it has hours,minutes,...

you can get a duration with hours > 24 by

absolute.difference(absolute)

in that case time.plus(duration) should wrap.

Reply:
Is Absolute an Instant?

Reply:
Yes. But it only balances up to hours, Since days/months/years are variable length units. can be problematic